nano/core_test/socket.cpp

```diff
@@ -41,22 +37,25 @@ TEST (socket, drop_policy)
 			return true;
 		});
 
-		auto client (std::make_shared<nano::socket> (*node, boost::none, nano::socket::concurrency::multi_writer));
+		auto client = std::make_shared<nano::socket> (*node, boost::none);
+		nano::transport::channel_tcp channel{ *node, client };
 		nano::util::counted_completion write_completion (total_message_count);
 
-		client->async_connect (boost::asio::ip::tcp::endpoint (boost::asio::ip::address_v4::loopback (), server_port),
-		[client, total_message_count, node, &write_completion, &drop_policy](boost::system::error_code const & ec_a) {
+		client->async_connect (boost::asio::ip::tcp::endpoint (boost::asio::ip::address_v6::loopback (), server_port),
+		[&channel, total_message_count, node, &write_completion, &drop_policy, client](boost::system::error_code const & ec_a) mutable {
 			for (int i = 0; i < total_message_count; i++)
 			{
 				std::vector<uint8_t> buff (1);
-				client->async_write (
-				nano::shared_const_buffer (std::move (buff)), [&write_completion](boost::system::error_code const & ec, size_t size_a) {
+				channel.send_buffer (
+				nano::shared_const_buffer (std::move (buff)), [&write_completion, client](boost::system::error_code const & ec, size_t size_a) mutable {
+					client.reset ();
 					write_completion.increment ();
 				},
 				drop_policy);
 			}
 		});
-		write_completion.await_count_for (std::chrono::seconds (5));
+		ASSERT_FALSE (write_completion.await_count_for (std::chrono::seconds (5)));
+		ASSERT_EQ (1, client.use_count ());
 	};
 
 	func (max_write_queue_size * 2 + 1, nano::buffer_drop_policy::no_socket_drop);
```

Comment:
Can it use `nano::socket::queue_size_max` directly? It seems `max_write_queue_size `is not used in other places